src/huggingface_hub/repocard.py

```diff
@@ -99,3 +104,153 @@ def metadata_eval_result(
         model_index, dict_factory=lambda x: {k: v for (k, v) in x if v is not None}
     )
     return {"model-index": [data]}
+
+
+def metadata_update(
+    repo_id: str,
+    metadata: Dict,
+    repo_type: str = None,
+    overwrite: bool = False,
+    token: str = None,
+) -> None:
+    """
+    Updates the metadata in the README.md of a repository on the Hugging Face Hub.
+
+    Args:
+        repo_id (`str`):
+            The name of the repository.
+        metadata (`dict`):
+            A dictionary containing the metadata to be updated.
+        repo_type (`str`, *optional*):
+            Set to `"dataset"` or `"space"` if updating to a dataset or space,
+            `None` or `"model"` if updating to a model. Default is `None`.
+        overwrite (`bool`, *optional*, defaults to `False`):
+            If set to `True` an existing field can be overwritten, otherwise
+            attempting to overwrite an existing field will cause an error.
+        token (`str`, *optional*):
+            The Hugging Face authentication token
+    """
+
+    filepath = hf_hub_download(
+        repo_id,
+        filename="README.md",
+        repo_type=repo_type,
+        use_auth_token=token,
+        force_download=True,
+    )
+    existing_metadata = metadata_load(filepath)
+
+    for key in metadata:
+        # update all fields except model index
+        if key != "model-index":
```

Comment:
Nit: I would have the logic the other way around. 

First `if key == "model-index", then else. 

It makes readability easier

Reply:
done